lightningd/plugin.c

```diff
@@ -1057,3 +1027,15 @@ void plugins_notify(struct plugins *plugins,
 	if (taken(n))
 		tal_free(n);
 }
+
+void plugin_request_send(struct plugin *plugin,
+			 struct jsonrpc_request *req TAKES)
+{
+	/* Add to map so we can find it later when routing the response */
+	tal_steal(plugin, req);
```

Comment:
This leaks take() if caller does it as implied by the TAKES above.  You can just document that this function consumed req, and remove the TAKES annotation?

Reply:
Hm, seems I'm missing the nuances of `TAKES` here, I thought the annotation means that it "takes ownership" and that the caller should not assume that the argument is accessible after the call anymore.

Reply:
It means "IF you call with take(x), it will take ownership".

Maybe we need an annotation for "consumes"?